source/fake.go

```diff
@@ -45,18 +43,19 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-// NewFakeSource creates a new fakeSource with the given client and namespace scope.
-func NewFakeSource(dnsName string) (Source, error) {
-	if dnsName == "" {
-		dnsName = defaultDNSName
+// NewFakeSource creates a new fakeSource with the given config.
+func NewFakeSource(cfg *Config) (Source, error) {
```

Comment:
I believe initialisers should not be changed. If I want to initialise just `FakeSource` I would have to pass it `source.Config` with many fields it doesn't need ?